src/HtmlParser.js

```diff
@@ -93,17 +97,30 @@ export default class HtmlParser {
       if (detect.hasOwnProperty(type)) {
         if (detect[type].test(this.stream)) {
           const token = streamReaders[type](this.stream);
+
           if (token) {
-            token.text = this.stream.substr(0, token.length);
-            return token;
+            if (token.type == 'startTag' &&
+                (/script|style/i).test(token.tagName)) {
```

Comment:
Yeah I know. We have that same code in HtmlParser and I considered those other options as well but none of the other ones are like script and style in that the others contain HTML within them and so could be parsed.